analytics/tests/test_counts.py

```diff
@@ -461,11 +462,12 @@ def setUp(self) -> None:
 
         # This realm should not show up in the *Count tables for any of the
         # messages_* CountStats
-        self.no_message_realm = Realm.objects.create(
-            string_id="no-message-realm",
-            name="No Message Realm",
-            date_created=self.TIME_ZERO - 2 * self.DAY,
+        self.no_message_realm = do_create_realm(
+            string_id="no-message-realm", name="No Message Realm"
         )
+        self.no_message_realm.date_created = self.TIME_ZERO - 2 * self.DAY
```

Comment:
It might be better to just make this an optional parameter, perhaps with an `assert self.TEST_SUITE` in that code path -- we do it in a lot of tests, and in general `realm.save(...)` in tests is an anti-pattern, for the same reason `Realm.objects.create` is.

Reply:
Makes sense, though I went with `assert not settings.PRODUCTION` because we use this in populate_analytics_db, which DEVELOPMENT, but not TEST_SUITE.